inference-engine/src/mkldnn_plugin/nodes/mkldnn_pooling_node.cpp

```diff
@@ -129,24 +144,16 @@ void MKLDNNPoolingNode::getSupportedDescriptors() {
     auto inputDataType = MKLDNNExtensionUtils::IEPrecisionToDataType(inputPrecision);
     auto outputDataType = MKLDNNExtensionUtils::IEPrecisionToDataType(outputPrecision);
 
-    effective_pad_begin = data_pad_begin;
-    effective_pad_end.resize(data_pad_end.size());
-
-    auto parentShape = getInputShapeAtPort(0);
-    auto childShape = getOutputShapeAtPort(0);
+    const auto &parentShape = getInputShapeAtPort(0);
+    const auto &childShape = getOutputShapeAtPort(0);
     const size_t inputRank = getInputShapeAtPort(0).getRank();
 
     if ((inputRank < 4) || (inputRank > 5))
         IE_THROW() << "Pooling layer. Unsupported mode. Only 4D and 5D blobs are supported as input.";
 
-    for (int i = 0; i < effective_pad_end.size(); i++) {
-        int krn = kernel[i];
-        int src = getInputShapeAtPort(0).getStaticDims()[2 + i];
-        int dst = getOutputShapeAtPort(0).getStaticDims()[2 + i];
+    initEffectivePad(MemoryDescUtils::makeDummyShape(parentShape),
```

Comment:
Why you call initEffectivePad twice: here and than inside CreateDescriptor()? Seems like one call is enough.